### PR TITLE
Thealmarty/cleanup

### PR DESCRIPTION
**Pull request by thealmarty**
- remove redundant imports
- apply linter suggestions 
- remove unused packages

Hopefully this will alleviate the out of RAM problem. We don't want to make it difficult for users to build Juvix.

### CODE REVIEW

File: test/Backends/Michelson.hs [hunk 1]
```diff
@@ -426,14 +367,9 @@ addDoublePairs =
...
-      J.Pi one (primTy int)
-        $ J.Pi one (primTy int)
-        $ primTy pairInt
-    xLook =
-      Ann one (primTy (Untyped.pair pairInt pairInt)) (J.Var "y")
-    applyPlus term name =
-      Ann one (primTy int) (J.AppM (addPairs name) [term])
+    pairIntType = J.Pi one (primTy int) $ J.Pi one (primTy int) $ primTy pairInt
+    xLook = Ann one (primTy (Untyped.pair pairInt pairInt)) (J.Var "y")
+    applyPlus term name = Ann one (primTy int) (J.AppM (addPairs name) [term])
```
Comment (mariari) [inline test/Backends/Michelson.hs:372]: I think this is much harder to read

File: test/Backends/Michelson.hs [hunk 2]
```diff
@@ -713,17 +621,12 @@ constUIntAns =
...
 xtwiceAns =
   [ PrimEx (PUSH "" (M.Type TInt "") (ValueInt 4)),
     PrimEx (PUSH "" (M.Type TInt "") (ValueInt 3)),
     PrimEx (PUSH "" (M.Type TInt "") (ValueInt 2)),
-    SeqEx
-      [ PrimEx (DIG 1),
-        PrimEx (DUP ""),
-        PrimEx (DUG 2)
-      ],
+    SeqEx [PrimEx (DIG 1), PrimEx (DUP ""), PrimEx (DUG 2)],
```
Comment (mariari) [inline test/Backends/Michelson.hs:629]: I believe these should stay as they were

File: test/Backends/Michelson.hs [hunk 3]
```diff
@@ -735,11 +638,7 @@ oddAppAns =
...
```
Comment (mariari) [inline test/Backends/Michelson.hs:641]: Ditto here

File: test/Backends/Michelson.hs [hunk 4]
```diff
@@ -602,9 +519,7 @@ identityAppTerm2 =
     $ J.LamM [] ["x"]
     $ Ann one (primTy (Untyped.pair opl Untyped.unit))
     $ J.AppM
-      ( Ann
-          one
-          (J.Pi one primPairTy (primTy (Untyped.pair opl Untyped.unit)))
+      ( Ann one (J.Pi one primPairTy (primTy (Untyped.pair opl Untyped.unit)))
```
Comment (mariari) [inline test/Backends/Michelson.hs:522]: I prefer the old style for this, as it allowed the arguments to align with the $

File: test/Backends/Michelson.hs [hunk 5]
```diff
@@ -629,18 +544,13 @@ identityAppTerm2 =
...
-    one
-    identityType2
+  Ann one identityType2
     $ J.LamM [] ["x"]
     $ Ann one (primTy (Untyped.pair opl Untyped.unit))
     $ J.AppM
-      ( Ann
-          one
-          (J.Pi one primPairTy (primTy (Untyped.pair unitl Untyped.unit)))
+      ( Ann one (J.Pi one primPairTy (primTy (Untyped.pair unitl Untyped.unit)))
```
Comment (mariari) [inline test/Backends/Michelson.hs:553]: I prefer the old style for this, as it allowed the arguments to align with the $

File: test/Backends/Michelson.hs [hunk 6]
```diff
@@ -449,18 +385,11 @@ xtwice =
           $ J.LamM [] ["y", "x", "z"]
           $ Ann one (primTy int)
           $ J.AppM
-            ( Ann
-                one
-                ( J.Pi one (primTy int)
-                    $ J.Pi one (primTy int)
-                    $ primTy int
-                )
+            ( Ann one (J.Pi one (primTy int) $ J.Pi one (primTy int) $ primTy int)
```
Comment (mariari) [inline test/Backends/Michelson.hs:388]: I  prefer the old style for this, as it allowed the arguments to align with the $

File: test/Backends/Michelson.hs [hunk 7]
```diff
@@ -449,18 +385,11 @@ xtwice =
...
                 $ J.Prim
                 $ Instructions.toNewPrimErr Instructions.mul
             )
-            [ Ann one (primTy int) (J.Var "x"),
-              Ann one (primTy int) (J.Var "x")
-            ]
+            [Ann one (primTy int) (J.Var "x"), Ann one (primTy int) (J.Var "x")]
```
Comment (mariari) [inline test/Backends/Michelson.hs:392]: I think this shows the vars less clear?

File: test/Backends/Michelson.hs [hunk 8]
```diff
@@ -237,24 +207,17 @@ pairConstant = pairGen [unitExpr1, unitExpr1]
...
-      ( Ann
-          one
-          ( J.Pi one (primTy Untyped.unit) $
-              primTy Untyped.unit
-          )
-          $ J.AppM
+  Ann one (primTy Untyped.unit) $
+    J.AppM
+      ( Ann one (J.Pi one (primTy Untyped.unit) $ primTy Untyped.unit) $
+          J.AppM
```
Comment (mariari) [inline test/Backends/Michelson.hs:220]: prefer the old style for this, as it allowed the arguments to align with the $

File: test/Backends/Michelson.hs [hunk 9]
```diff
@@ -194,31 +174,21 @@ constUInt =
...
-    (primTy Untyped.unit)
-    $ J.AppM constUInt [unitExpr1, annIntOne 3]
+  Ann one (primTy Untyped.unit) $ J.AppM constUInt [unitExpr1, annIntOne 3]
 
 pairGen :: [AnnTerm PrimTy NewPrim] -> AnnTerm PrimTy NewPrim
 pairGen =
-  Ann
-    one
-    (primTy (Untyped.pair Untyped.unit Untyped.unit))
+  Ann one (primTy (Untyped.pair Untyped.unit Untyped.unit))
```
Comment (mariari) [inline test/Backends/Michelson.hs:191]: prefer the old style for this, as it allowed the arguments to align with the $

File: src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs [hunk 10]
```diff
@@ -316,16 +315,17 @@ onTwoArgs op f typ instrs = do
...
+          then
+            let Env.Constant i1 = val instr1
+                Env.Constant i2 = val instr2
+             in pure (f i1 i2)
+          else do
+            traverse_ addExpanded instrs
+            -- add when we normalize
+            -- copyAndDrop 2
+            addInstr op
+            pure Env.Nop
```
Comment (mariari) [inline src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs:328]: Best to keep the |, makes it more like a cond form than an if form. VIsually cleaner IMO

**Comment by cwgoes** [comment]
With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want specifically).

**Comment by mariari** [comment]
> With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want only).

the linter suggested this? The code compiled as is in the code base is formatted by Ormulo already

**Comment by thealmarty** [comment]
> > With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want only).
> 
> the linter suggested this? The code compiled as is in the code base is formatted by Ormulo already

Yes, those are suggestions from the formatter or the linter. I agree with Chris, I don't think we should argue about personal opinion on the code. And I prefer to not change the linter.

**Comment by mariari** [comment]
> > > With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want only).
> > 
> > 
> > the linter suggested this? The code compiled as is in the code base is formatted by Ormulo already
> 
> Yes, those are suggestions from the formatter or the linter. I agree with Chris, I don't think we should argue about personal opinion on the code. And I prefer to not change the linter.

What linter is this? The only tool I run is Ormulo, which does not make this change

**Comment by thealmarty** [comment]
> > > > With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want only).
> > > 
> > > 
> > > the linter suggested this? The code compiled as is in the code base is formatted by Ormulo already
> > 
> > 
> > Yes, those are suggestions from the formatter or the linter. I agree with Chris, I don't think we should argue about personal opinion on the code. And I prefer to not change the linter.
> 
> What linter is this? The only tool I run is Ormulo, which does not make this change

HLint (see our [style guide](https://github.com/cryptiumlabs/juvix/blob/develop/doc/STYLEGUIDE.md)).

**Comment by mariari** [comment]
> > > > > With regards to the formatting questions, can we either accept the linter suggestions or change the linter? I don't think it makes sense to use a linter if we don't accept its suggestions (or configure it to do what we want only).
> > > > 
> > > > 
> > > > the linter suggested this? The code compiled as is in the code base is formatted by Ormulo already
> > > 
> > > 
> > > Yes, those are suggestions from the formatter or the linter. I agree with Chris, I don't think we should argue about personal opinion on the code. And I prefer to not change the linter.
> > 
> > 
> > What linter is this? The only tool I run is Ormulo, which does not make this change
> 
> HLint (see our [style guide](https://github.com/cryptiumlabs/juvix/blob/develop/doc/STYLEGUIDE.md)).

Can you show me where it generates these actions?

I get the following
```bash
[redacted]: hlint -report test/Backends/Michelson.hs 
test/Backends/Michelson.hs:944:34: Warning: Redundant bracket
Found:
  (M.ValueNil)
Perhaps:
  M.ValueNil

Writing report to eport ...
1 hint
```

**Comment by thealmarty** [comment]
> Can you show me where it generates these actions?
> 
> I get the following
> 
> ```shell
> [redacted]: hlint -report test/Backends/Michelson.hs 
> test/Backends/Michelson.hs:944:34: Warning: Redundant bracket
> Found:
>   (M.ValueNil)
> Perhaps:
>   M.ValueNil
> 
> Writing report to eport ...
> 1 hint
> ```
> 
> It seems for the other file it has a parse error

Oops, I missed that one. I'll get that too. Can I merge after?

**Comment by mariari** [comment]
> > Can you show me where it generates these actions?
> > I get the following
> > ```shell
> > [redacted]: hlint -report test/Backends/Michelson.hs 
> > test/Backends/Michelson.hs:944:34: Warning: Redundant bracket
> > Found:
> >   (M.ValueNil)
> > Perhaps:
> >   M.ValueNil
> > 
> > Writing report to eport ...
> > 1 hint
> > ```
> > 
> > 
> > It seems for the other file it has a parse error
> 
> Oops, I missed that one. I'll get that too. Can I merge after?



> > Can you show me where it generates these actions?
> > I get the following
> > ```shell
> > [redacted]: hlint -report test/Backends/Michelson.hs 
> > test/Backends/Michelson.hs:944:34: Warning: Redundant bracket
> > Found:
> >   (M.ValueNil)
> > Perhaps:
> >   M.ValueNil
> > 
> > Writing report to eport ...
> > 1 hint
> > ```
> > 
> > 
> > It seems for the other file it has a parse error
> 
> Oops, I missed that one. I'll get that too. Can I merge after?

This is from my mariari/contract-generation branch. How did you get those other changes, it seems hlint does not warn about them

**Comment by thealmarty** [comment]
Oic, I didn't miss that. I don't know why yours doesn't show. Some of them may be the formatter too. What version of HLint are you using? I can get us synced up but let's move this discussion to another channel.

**Pull request merged**